sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +854,298 @@ def test_robust_scaler_iris_quantiles():
     assert_array_almost_equal(q_range, 1)
 
 
+def test_quantile_transform_iris():
+    X = iris.data
+    transformer = QuantileTransformer(n_quantiles=30)
+    X_trans = transformer.fit_transform(X)
+    assert_array_almost_equal(np.min(X_trans, axis=0), 0.)
+    assert_array_almost_equal(np.max(X_trans, axis=0), 1.)
+    X_trans_inv = transformer.inverse_transform(X_trans)
+    assert_array_almost_equal(X, X_trans_inv)
+
+
+def test_quantile_transform_check_error():
+    X = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                  [2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                  [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X = sparse.csc_matrix(X)
+    X_neg = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                      [-2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                      [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X_neg = sparse.csc_matrix(X_neg)
+
+    assert_raises_regex(ValueError, "Invalid value for 'n_quantiles'",
+                        QuantileTransformer(n_quantiles=0).fit, X_neg)
```

Comment:
I don't get why you're running these tests with `X_neg` rather than `X`. Surely we're not so picky about which validation is done first that we make a test of it.